rcl/src/rcl/arguments.c

```diff
@@ -267,6 +301,26 @@ rcl_parse_arguments(
         continue;
       }
 
+      // Attempt to parse argument as parameter override flag
+      if (strcmp(RCL_PARAM_FLAG, argv[i]) == 0 || strcmp(RCL_SHORT_PARAM_FLAG, argv[i]) == 0) {
+        if (i + 1 < argc) {
+          // Attempt to parse next argument as parameter override rule
+          if (RCL_RET_OK == _rcl_parse_param_rule(argv[i + 1], args_impl->parameter_overrides)) {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "param override rule : %s\n", argv[i + 1]);
+            i += 1;  // Skip flag here, for loop will skip rule.
+            continue;
+          } else {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
```

Comment:
Not entirely sure, thus [my comment above](https://github.com/ros2/rcl/pull/483#discussion_r312532781). Currently, `rcl_parse_arguments` has no concept of a bad argument, it simply avoids acknowledging them as such and defers error handling to upper layers (as it has to). I kept it as-is to avoid having to define how it should behave (i.e. how it notifies the caller about a bad option, what it does with the half built `rcl_arguments` struct, etc.). Also, it allows upper layers to give other meanings to `rcl` flags in their own custom ways, which I personally think is not a great idea but others may disagree.

Reply:
Now that we have the concept of `--ros-args`, I think we should have a concept of bad argument here. It's also ok to have an array with non parsed ros specific arguments, because we may want to have specific options for only one client library and not the other.
But IMO, if a remapping rule was badly written we should notify the error here and not defer it .
I think we can left this discussion for a follow-up if you want.